src/daemon/daemon.cpp

```diff
@@ -595,9 +595,9 @@ mp::Daemon::Daemon(std::unique_ptr<const DaemonConfig> the_config)
             config->vault->remove(name);
         }
 
-        if (spec.state == VirtualMachine::State::running && vm_instances[name]->state != VirtualMachine::State::running)
+        if (spec.state == VirtualMachine::State::running && !spec.deleted &&
```

Comment:
I thought this a bit more over at lunch. If we're getting inconsistent state from disk, I think we should log and correct, (to stopped or unknown).

```c++
if(spec.state == VirtualMachine::State::running)
    if(spec.deleted)
    {
        // log
        spec.state = VirtualMachine::State::unknown;
    }
    else if(vm_instances[name]->state != VirtualMachine::State::running)
    {
        [...]
    }